### ISSUE TITLE
Unexpected behaviour of GraphQL/PupQL subscription

### ISSUE THREAD

**Issue by rlhk** [cleverbeagle/pup#273]
I've been away from Meteor for a while and quite new to the GraphQL setup.

I'm playing on the comment subscription setup of a new Pup project. And I observed the following unexpected behaviours:

1. Opened a few Chrome tabs/windows (one is Incognito mode), plus a mobile Safari (related problem solved in another issue: https://github.com/cleverbeagle/pup/issues/271). Let's say session A,B,C
2. All sessions logged in as admin
3. All sessions browse to same document (the 1st test document in my tests)
4. Now try add comment from any session.

To my surprise, ONLY comments list of ONE session seems to be synced.
* Say, when I refresh session A, add a comment, the comments list updates in A only
* I could add comment from session B, but only sees comments list update in A only
* Only after I refresh session B, adding of comment from A, B, C shows up in only B but not other sessions.
* Refreshing session A sets the update to A again. etc.
* By the way, after a while, the console begins to show `Network error: Failed to fetch at new ApolloError` ... in the consoles of all sessions.

I tried many times and still have the same issue.

Is this a known issue? Is it a Meteor or PupQL issue?

**Comment by cleverbeagle** [comment]
Hi @rlhk, good question. I noticed this the other day but was hesitant to log an issue. My bet would be something with the WebSockets implementation—I hadn't caught that `Network error` you posted, so I'll need to take a look.

**Comment by jesperordrup** [comment]
Confirmed. Ive seen similar behaviour. Here's a screenshot from console:

![image](https://user-images.githubusercontent.com/3120151/48728264-ca392f80-ec34-11e8-92b6-0d219edae207.png)

**Comment by johnsonchandra** [comment]
any update on this issue?

I noticed that on server it triggered the onDisconnect at SubscriptionServer.create  
onDisconnect: () => {
      console.log(`[PupQL] Woof! Subscription client disconnected...`);
    }

why would it be disconnected for previous connections, and only the last client is keep connected? 
any idea guys?

**Comment by cleverbeagle** [comment]
@johnsonchandra it's on my radar but haven't taken a look yet.

**Comment by tnedich** [comment]
Good evening, I have same problem. Also, when record is deleted from MongoDB there is no automatic update od UI. It creates some strange situation where I can see record (Document for example) that is non existant in database. Only "hard" refresh helps.

**Comment by cleverbeagle** [comment]
@tnedich an example of how to temporarily resolve this is available here: https://github.com/cleverbeagle/pup/issues/284#issuecomment-487780775.

My plan is to remove PupQL and handle subscriptions via another means (I'm not happy with the Apollo implementation and have some ideas for how to improve it).

**Comment by danyhiol** [comment]
> Confirmed. Ive seen similar behaviour. Here's a screenshot from console:
> 
> ![image](https://user-images.githubusercontent.com/3120151/48728264-ca392f80-ec34-11e8-92b6-0d219edae207.png)

I'm having a similar issue when performing more than 3 queries.
I need to perform 4 to 5 queries in a single. But when performing more than 2 queries, the last query gets stuck on `loading=true` and return `[Network error]: TypeError: Failed to fetch`.
I tried out the temporal fix mention above but the error still persists when performing more than 3 queries.

**Comment by cleverbeagle** [comment]
@danyhiol the `loading: true` issue is a bug in Apollo that's come up a few times for other folks. The hack I've done to handle this is to automatically perform a refetch on `componentDidMount` or another lifecycle method (if your query is dependent on data coming through).

**Comment by danyhiol** [comment]
@cleverbeagle you are right. I've been facing this problem for some days now. And no solution actually worked for me. I tried your solution using `react-hooks` (`useEffect`) since my app is refracted to `react functions` with no success. I end up with a temporary solution wish consist of using `setTimeout` functions to `execute refetch()` when `loading` is `true`. 
I would like to know more about your solution if possible using` react-hooks`. Since even after `componentDiMout`, `loading` is still `true` wish leads to a `loop` or to the same old problem.

**Comment by cleverbeagle** [comment]
Closed in favor of #284. Will likely move to a non-Apollo solution for subscriptions/real-time in the future.
